test-case-reporting/src/setup_db.ts

```diff
@@ -44,13 +44,14 @@ export async function setupDb(db: Database): Promise<unknown> {
     .createTable('test_cases', table => {
       // MD5 hash of the name
       table
-        .string('id', 32)
+        .specificType('id', 'char(32)')
+        .notNullable()
```

Comment:
I think `primary` implies `notNullable`, double check knex docs though

Reply:
See http://knexjs.org/#Schema-primary.
> On Amazon Redshift, all columns included in a primary key must be not nullable.

This suggests that the primary key could be nullable in other database types.